### PR TITLE
Issue538

### PR DESCRIPTION
**Pull request by AnjaLeBlanc**
#538 

Modified the two shell scripts 'goostats' and 'goodiff' by checking input arguments and parameters. When using them with the wrong number of arguments the scripts produce some help text.
Added a couple of lines to the story about the scripts. (As the don't do anything useful without too much detail!) 

(Pull request as part of Instructor training checkout process)

### CODE REVIEW

File: _episodes/06-script.md [hunk 1]
```diff
@@ -320,6 +320,9 @@ The file `redo-figure-3.sh` now contains:
 ~~~
 {: .source}
 
+
+(goostats is a shell script which calculates some complicated statistics from a datafile -- the first parameter -- and writes them to a file -- the second parameter; the script goodiff compares two datafiles provided as parameters. Nele's superviser provided them without too many explainations.)
```
Comment (gcapes) [inline _episodes/06-script.md]: Please change ~parameter~ argument. We've just tidied up the usage of argument vs parameter in the shell lesson #624.
Could you also use backticks for `goostats` and `goodiff`.
And fix ~Nele's~ Nelle's

**Comment by shwina** [comment]
Perhaps this is a good opportunity to get rid entirely of the optional arguments in `goostats`. We don't use them in the lesson as far as I'm aware. @AnjaLeBlanc, @gcapes what do you think?

**Comment by AnjaLeBlanc** [comment]
@shwina -- the optional args don't do anything but they are mentioned in the story line (Chapter 6 -- Nelle’s Pipeline: Creating a Script). But I agree they are not essential and could shorten the lesson by a few seconds.

**Comment by shwina** [comment]
Thanks @AnjaLeBlanc! Removing them also has the added benefit of making the `goostats` script actually readable. Often after the exercise I will open the script up to explain what it is doing (i.e., a few operations like `uniq` and `cut` that we have seen before and then sleeping for 2 seconds).

I'd be in favour of removing them entirely.

**Comment by gcapes** [comment]
As Anja mentioned, the extra flags are the justification given in the story as to why Nelle needs to re-run the script. 

If we remove this part of the story, we're re-running the script for no reason. I think we need something in the story to justify re-running the script - those extra flags were ideal. So my vote is to keep the flags, or remove them and add a different credible justification for re-running the script.

**Comment by shwina** [comment]
@gcapes Thanks! In practice I do not use these switches. Instead I will ask learners to imagine they wille obtain a few more files the next day and will have to re-do the analysis: having a script can avoid re-typing the same commands.

@gcapes, if you think that's a good enough motivation, @AnjaLeBlanc, could I please request you to add it as part of the lesson narrative?

@AnjaLeBlanc thanks for all your contribution so far!

**Comment by gcapes** [comment]
@shwina The problem with that reasoning is that you would only want to process the new files rather than reprocess the old files too (so you're implicitly suggesting bad practice to the learners). It doesn't seem a good enough justification - sorry! I think the original story should stay.

File: data-shell/north-pacific-gyre/2012-07-03/goostats [hunk 2]
```diff
@@ -1,24 +1,8 @@
...
-done
-
-shift $((OPTIND-1))
-
 # check for the right number of input arguments
 if [ $# -ne 2 ]
   then
-    echo "goodstat [-J number] [-r] file1 file2"
-    echo "call goostat with two arguments it accepts the optional options -J with a number and -r"
+    echo "goodstat file1 file2"
```
Comment (shwina) [inline data-shell/north-pacific-gyre/2012-07-03/goostats]: `goodstat` -> `goostats`

File: _episodes/06-script.md [hunk 3]
```diff
@@ -312,14 +312,17 @@ $ history | tail -n 5 > redo-figure-3.sh
...
+297 bash goostats NENE01729B.txt stats-NENE01729B.txt
 298 bash goodiff stats-NENE01729B.txt /data/validated/01729.txt > 01729-differences.txt
 299 cut -d ',' -f 2-3 01729-differences.txt > 01729-time-series.txt
 300 ygraph --format scatter --color bw --borders none 01729-time-series.txt figure-3.png
 301 history | tail -n 5 > redo-figure-3.sh
 ~~~
 {: .source}
 
+
+(`goostats` is a shell script which calculates some complicated statistics from a datafile -- the first argument -- and writes them to a file -- the second argument; the script `goodiff` compares two datafiles provided as arguments. Nelle's superviser provided them without too many explainations.)
```
Comment (shwina) [inline _episodes/06-script.md]: 1. I suggest renaming datafile (not a word) to either "input file", "file" or "protein sample file".

2. I also suggest breaking up the first sentence into two, as `goostats` and `goodiff` are sufficiently different scripts. `goodiff` is not supposedly comparing files, but rather "Compar[ing] the statistics for each protein with corresponding statistics for each other protein ". I would re-word as "`goodiff` is a script for comparing statistics between two files, provided as arguments".

3. A few small typos: superviser->supervisor, explainations->explanations

File: data-shell/north-pacific-gyre/2012-07-03/goostats [hunk 4]
```diff
@@ -1,11 +1,10 @@
...
-    case "$option" in
-    J) Jarg="$OPTARG"
-       shift $((OPTIND));;
-    r) shift $((OPTIND-1));;
-    esac
-done
+# check for the right number of input arguments
+if [ $# -ne 2 ]
+  then
+    echo "goodstat file1 file2"
```
Comment (shwina) [inline data-shell/north-pacific-gyre/2012-07-03/goostats]: Small typo: `goodstat` -> `goostats`

Comment (AnjaLeBlanc) [reply data-shell/north-pacific-gyre/2012-07-03/goostats]: @gcapes 
How about this justification of the script: 
Nelle's supervisor insisted that all her analytics must be reproducible. The easiest way to capture all the steps is in a script.

**Comment by AnjaLeBlanc** [comment]
@gcapes
How about this justification of the script:
Nelle's supervisor insisted that all her analytics must be reproducible. The easiest way to capture all the steps is in a script.

**Comment by gcapes** [comment]
Yep, that sounds perfect!

**Comment by shwina** [comment]
Looks good to me! @gcapes any suggestions before merge?

**Comment by gcapes** [comment]
Merge, merge, merge! :)

**Comment by shwina** [comment]
Thanks @AnjaLeBlanc for all your work on this!

**Comment by gdevenyi** [comment]
Wonderful work, I appreciate the careful coordination of edits to both the code and the narrative.

**Pull request merged**